src/libraries/System.Data.Odbc/src/Common/System/Data/ProviderBase/DbBuffer.cs

```diff
@@ -659,121 +767,162 @@ internal void ZeroMemory()
             }
         }
 
-        internal Guid ReadGuid(int offset)
+        internal unsafe Guid ReadGuid(int offset)
         {
-            // faster than Marshal.PtrToStructure(offset, typeof(Guid))
-            byte[] buffer = new byte[16];
-            ReadBytes(offset, buffer, 0, 16);
-            return new Guid(buffer);
+            offset += BaseOffset;
+
+            Validate(offset, sizeof(Guid));
+
+            Debug.Assert(0 == offset % /* alignof(Guid) */ sizeof(int), "invalid alignment");
```

Comment:
@roji, @ajcvickers Note for code owners, this change relaxes the alignment requirement from `ADP.PtrSize`. I believe this is acceptable, but it is still a change in behavior.

please confirm tthe alignment requirement from `IntPtr` is acceptable.